crates/rspack_regex/src/lib.rs

```diff
@@ -10,6 +12,8 @@ mod algo;
 /// Using wrapper type required by [TryFrom] trait
 #[derive(Debug, Clone)]
 pub struct RspackRegex {
+  pub expr: String,
+  pub flags: String,
   pub algo: Algo,
```

Comment:
I thought out another way for hashing `Algo`, you can define a `AlgoHash` trait and implement it for `Algo`, then use the `AlgoHash` when calling `<RspackRegex as Hash>::hash`, this avoid clone caused by `to_string`

Reply:
Trait can not has its own fields and Algo is an enum. So I add a wrapper struct named HashRegressRegex which implement the hash function and overwrite the default Debug::fmt function.